app/src/main/res/layout/activity_main.xml

```diff
@@ -59,14 +59,49 @@
                 android:textAppearance="@style/TextAppearance.AppCompat.Medium" />
         </LinearLayout>
 
-        <!-- TODO (6) Add a LinearLayout that contains a checkbox to grant the location permissions
-                        and set the checkbox click listener to call onLocationPermissionClicked -->
+        <LinearLayout
```

Comment:
One little optimization.
We can use just a Checkbox without any Layout

<CheckBox
            android:id="@+id/location_permission_checkbox"
            android:layout_width="match_parent"
            android:layout_height="wrap_content"
            android:text="@string/checkbox_location_permission"
            android:textAppearance="@style/TextAppearance.AppCompat.Medium"
            android:layout_marginTop="12dp"
            android:button="@null"
            android:drawablePadding="8dp"
            android:drawableLeft="@drawable/ic_my_location_primary_24dp"
            android:drawableStart="@drawable/ic_my_location_primary_24dp"
            android:drawableRight="?android:attr/listChoiceIndicatorMultiple"
            android:drawableEnd="?android:attr/listChoiceIndicatorMultiple"
            android:onClick="onLocationPermissionClicked"/>